From b491537117f89dc781aa49d04c7bf87dbe36978b Mon Sep 17 00:00:00 2001
From: Randolph <randolph@andestech.com>
Date: Wed, 21 Aug 2024 17:34:34 +0800
Subject: [PATCH 19/53] XIP: new Flash mapping (#117)

u-boot-spl: extend the flash space to 256 KB
u-boot.itb: start at 0x40000, extend the flash space to 896 KB.
DTB: start at 0x1E0000, extend the flash space to 64 KB.
U-Boot Env.: start at 0x1F0000, extend the flash space to 64 KB.

Modify CONFIG_SPL_MAX_SIZE to match the flash size.

Signed-off-by: Randolph <randolph@andestech.com>
Reviewed-on: https://gitea.andestech.com/RD-SW/u-boot.official/pulls/117
Reviewed-by: Mina Hui-Min Chou <minachou@andestech.com>
Reviewed-by: Leo Yu-Chi Liang <ycliang@andestech.com>
Reviewed-by: Tim Shih-Ting OuYang <tim609@andestech.com>
Co-authored-by: Randolph <randolph@andestech.com>
Co-committed-by: Randolph <randolph@andestech.com>
---
 board/andestech/ae350/Kconfig           | 4 ++--
 configs/ae350_rv32_falcon_defconfig     | 4 ++--
 configs/ae350_rv32_falcon_xip_defconfig | 4 ++--
 configs/ae350_rv32_spl_defconfig        | 4 ++--
 configs/ae350_rv32_spl_xip_defconfig    | 6 +++---
 configs/ae350_rv64_falcon_defconfig     | 4 ++--
 configs/ae350_rv64_falcon_xip_defconfig | 6 +++---
 configs/ae350_rv64_spl_defconfig        | 4 ++--
 configs/ae350_rv64_spl_xip_defconfig    | 6 +++---
 9 files changed, 21 insertions(+), 21 deletions(-)

diff --git a/board/andestech/ae350/Kconfig b/board/andestech/ae350/Kconfig
index 096564b3dc16..619363fc3b48 100644
--- a/board/andestech/ae350/Kconfig
+++ b/board/andestech/ae350/Kconfig
@@ -19,7 +19,7 @@ config ENV_SIZE
 	default 0x2000 if ENV_IS_IN_SPI_FLASH
 
 config ENV_OFFSET
-	default 0x140000 if ENV_IS_IN_SPI_FLASH
+	default 0x1F0000 if ENV_IS_IN_SPI_FLASH
 
 config SPL_TEXT_BASE
 	default 0x800000
@@ -29,7 +29,7 @@ config SPL_OPENSBI_LOAD_ADDR
 
 config SYS_FDT_BASE
 	hex
-	default 0x800f0000 if OF_SEPARATE
+	default 0x801E0000 if OF_SEPARATE
 
 config BOARD_SPECIFIC_OPTIONS # dummy
 	def_bool y
diff --git a/configs/ae350_rv32_falcon_defconfig b/configs/ae350_rv32_falcon_defconfig
index ea2205d8ea47..53952589163f 100644
--- a/configs/ae350_rv32_falcon_defconfig
+++ b/configs/ae350_rv32_falcon_defconfig
@@ -25,7 +25,7 @@ CONFIG_SYS_PBSIZE=1050
 CONFIG_DISPLAY_CPUINFO=y
 CONFIG_DISPLAY_BOARDINFO=y
 CONFIG_BOARD_EARLY_INIT_R=y
-CONFIG_SPL_MAX_SIZE=0x100000
+CONFIG_SPL_MAX_SIZE=0x40000
 CONFIG_SPL_BOARD_INIT=y
 CONFIG_SPL_CACHE=y
 CONFIG_SPL_PAYLOAD_ARGS_ADDR=0x01700000
@@ -61,4 +61,4 @@ CONFIG_SPI=y
 CONFIG_ATCSPI200_SPI=y
 # CONFIG_BINMAN_FDT is not set
 CONFIG_CONSOLE_RECORD=y
-CONFIG_CONSOLE_RECORD_OUT_SIZE_F=0x2000
\ No newline at end of file
+CONFIG_CONSOLE_RECORD_OUT_SIZE_F=0x2000
diff --git a/configs/ae350_rv32_falcon_xip_defconfig b/configs/ae350_rv32_falcon_xip_defconfig
index 5c6d43f01970..150863846072 100644
--- a/configs/ae350_rv32_falcon_xip_defconfig
+++ b/configs/ae350_rv32_falcon_xip_defconfig
@@ -18,7 +18,7 @@ CONFIG_SPL_XIP=y
 CONFIG_SPL_LOAD_FIT_OPENSBI_OS_BOOT=y
 CONFIG_SYS_MONITOR_BASE=0x88000000
 CONFIG_FIT=y
-CONFIG_SPL_LOAD_FIT_ADDRESS=0x80010000
+CONFIG_SPL_LOAD_FIT_ADDRESS=0x80040000
 CONFIG_SYS_BOOTM_LEN=0x4000000
 CONFIG_DISTRO_DEFAULTS=y
 CONFIG_BOOTDELAY=3
@@ -26,7 +26,7 @@ CONFIG_SYS_PBSIZE=1050
 CONFIG_DISPLAY_CPUINFO=y
 CONFIG_DISPLAY_BOARDINFO=y
 CONFIG_BOARD_EARLY_INIT_R=y
-CONFIG_SPL_MAX_SIZE=0x100000
+CONFIG_SPL_MAX_SIZE=0x40000
 CONFIG_SPL_BOARD_INIT=y
 CONFIG_SPL_CACHE=y
 CONFIG_SPL_PAYLOAD_ARGS_ADDR=0x01700000
diff --git a/configs/ae350_rv32_spl_defconfig b/configs/ae350_rv32_spl_defconfig
index fdaa25988188..4b4caa0032ed 100644
--- a/configs/ae350_rv32_spl_defconfig
+++ b/configs/ae350_rv32_spl_defconfig
@@ -26,7 +26,7 @@ CONFIG_DISPLAY_CPUINFO=y
 CONFIG_DISPLAY_BOARDINFO=y
 CONFIG_BOARD_EARLY_INIT_R=y
 CONFIG_MISC_INIT_R=y
-CONFIG_SPL_MAX_SIZE=0x100000
+CONFIG_SPL_MAX_SIZE=0x40000
 CONFIG_SPL_BOARD_INIT=y
 CONFIG_SPL_CACHE=y
 CONFIG_SPL_OPENSBI_SCRATCH_OPTIONS=0x0
@@ -65,4 +65,4 @@ CONFIG_WDT=y
 CONFIG_WDT_ATCWDT200=y
 # CONFIG_BINMAN_FDT is not set
 CONFIG_CONSOLE_RECORD=y
-CONFIG_CONSOLE_RECORD_OUT_SIZE_F=0x2000
\ No newline at end of file
+CONFIG_CONSOLE_RECORD_OUT_SIZE_F=0x2000
diff --git a/configs/ae350_rv32_spl_xip_defconfig b/configs/ae350_rv32_spl_xip_defconfig
index e5beea04a86b..edf3c3156807 100644
--- a/configs/ae350_rv32_spl_xip_defconfig
+++ b/configs/ae350_rv32_spl_xip_defconfig
@@ -18,7 +18,7 @@ CONFIG_RISCV_SMODE=y
 CONFIG_SPL_XIP=y
 CONFIG_SYS_MONITOR_BASE=0x88000000
 CONFIG_FIT=y
-CONFIG_SPL_LOAD_FIT_ADDRESS=0x80010000
+CONFIG_SPL_LOAD_FIT_ADDRESS=0x80040000
 CONFIG_SYS_BOOTM_LEN=0x4000000
 CONFIG_DISTRO_DEFAULTS=y
 CONFIG_BOOTDELAY=3
@@ -27,7 +27,7 @@ CONFIG_DISPLAY_CPUINFO=y
 CONFIG_DISPLAY_BOARDINFO=y
 CONFIG_BOARD_EARLY_INIT_R=y
 CONFIG_MISC_INIT_R=y
-CONFIG_SPL_MAX_SIZE=0x100000
+CONFIG_SPL_MAX_SIZE=0x40000
 CONFIG_SPL_BOARD_INIT=y
 CONFIG_SPL_CACHE=y
 CONFIG_SPL_OPENSBI_SCRATCH_OPTIONS=0x0
@@ -66,4 +66,4 @@ CONFIG_WDT=y
 CONFIG_WDT_ATCWDT200=y
 # CONFIG_BINMAN_FDT is not set
 CONFIG_CONSOLE_RECORD=y
-CONFIG_CONSOLE_RECORD_OUT_SIZE_F=0x2000
\ No newline at end of file
+CONFIG_CONSOLE_RECORD_OUT_SIZE_F=0x2000
diff --git a/configs/ae350_rv64_falcon_defconfig b/configs/ae350_rv64_falcon_defconfig
index 0d229207dfc9..a984ce51a70f 100644
--- a/configs/ae350_rv64_falcon_defconfig
+++ b/configs/ae350_rv64_falcon_defconfig
@@ -25,7 +25,7 @@ CONFIG_SYS_PBSIZE=1050
 CONFIG_DISPLAY_CPUINFO=y
 CONFIG_DISPLAY_BOARDINFO=y
 CONFIG_BOARD_EARLY_INIT_R=y
-CONFIG_SPL_MAX_SIZE=0x100000
+CONFIG_SPL_MAX_SIZE=0x40000
 CONFIG_SPL_BOARD_INIT=y
 CONFIG_SPL_CACHE=y
 CONFIG_SPL_PAYLOAD_ARGS_ADDR=0x01700000
@@ -61,4 +61,4 @@ CONFIG_SPI=y
 CONFIG_ATCSPI200_SPI=y
 # CONFIG_BINMAN_FDT is not set
 CONFIG_CONSOLE_RECORD=y
-CONFIG_CONSOLE_RECORD_OUT_SIZE_F=0x2000
\ No newline at end of file
+CONFIG_CONSOLE_RECORD_OUT_SIZE_F=0x2000
diff --git a/configs/ae350_rv64_falcon_xip_defconfig b/configs/ae350_rv64_falcon_xip_defconfig
index 8834aa0b05e8..7f2d1a74594f 100644
--- a/configs/ae350_rv64_falcon_xip_defconfig
+++ b/configs/ae350_rv64_falcon_xip_defconfig
@@ -18,7 +18,7 @@ CONFIG_SPL_XIP=y
 CONFIG_SPL_LOAD_FIT_OPENSBI_OS_BOOT=y
 CONFIG_SYS_MONITOR_BASE=0x88000000
 CONFIG_FIT=y
-CONFIG_SPL_LOAD_FIT_ADDRESS=0x80010000
+CONFIG_SPL_LOAD_FIT_ADDRESS=0x80040000
 CONFIG_SYS_BOOTM_LEN=0x4000000
 CONFIG_DISTRO_DEFAULTS=y
 CONFIG_BOOTDELAY=3
@@ -26,7 +26,7 @@ CONFIG_SYS_PBSIZE=1050
 CONFIG_DISPLAY_CPUINFO=y
 CONFIG_DISPLAY_BOARDINFO=y
 CONFIG_BOARD_EARLY_INIT_R=y
-CONFIG_SPL_MAX_SIZE=0x100000
+CONFIG_SPL_MAX_SIZE=0x40000
 CONFIG_SPL_BOARD_INIT=y
 CONFIG_SPL_CACHE=y
 CONFIG_SPL_PAYLOAD_ARGS_ADDR=0x01700000
@@ -62,4 +62,4 @@ CONFIG_SPI=y
 CONFIG_ATCSPI200_SPI=y
 # CONFIG_BINMAN_FDT is not set
 CONFIG_CONSOLE_RECORD=y
-CONFIG_CONSOLE_RECORD_OUT_SIZE_F=0x2000
\ No newline at end of file
+CONFIG_CONSOLE_RECORD_OUT_SIZE_F=0x2000
diff --git a/configs/ae350_rv64_spl_defconfig b/configs/ae350_rv64_spl_defconfig
index 7a7a628ed578..03739935d234 100644
--- a/configs/ae350_rv64_spl_defconfig
+++ b/configs/ae350_rv64_spl_defconfig
@@ -26,7 +26,7 @@ CONFIG_DISPLAY_CPUINFO=y
 CONFIG_DISPLAY_BOARDINFO=y
 CONFIG_BOARD_EARLY_INIT_R=y
 CONFIG_MISC_INIT_R=y
-CONFIG_SPL_MAX_SIZE=0x100000
+CONFIG_SPL_MAX_SIZE=0x40000
 CONFIG_SPL_BOARD_INIT=y
 CONFIG_SPL_CACHE=y
 CONFIG_SPL_OPENSBI_SCRATCH_OPTIONS=0x0
@@ -65,4 +65,4 @@ CONFIG_WDT=y
 CONFIG_WDT_ATCWDT200=y
 # CONFIG_BINMAN_FDT is not set
 CONFIG_CONSOLE_RECORD=y
-CONFIG_CONSOLE_RECORD_OUT_SIZE_F=0x2000
\ No newline at end of file
+CONFIG_CONSOLE_RECORD_OUT_SIZE_F=0x2000
diff --git a/configs/ae350_rv64_spl_xip_defconfig b/configs/ae350_rv64_spl_xip_defconfig
index e56cd7cb19e5..daa133d09cad 100644
--- a/configs/ae350_rv64_spl_xip_defconfig
+++ b/configs/ae350_rv64_spl_xip_defconfig
@@ -18,7 +18,7 @@ CONFIG_RISCV_SMODE=y
 CONFIG_SPL_XIP=y
 CONFIG_SYS_MONITOR_BASE=0x88000000
 CONFIG_FIT=y
-CONFIG_SPL_LOAD_FIT_ADDRESS=0x80010000
+CONFIG_SPL_LOAD_FIT_ADDRESS=0x80040000
 CONFIG_SYS_BOOTM_LEN=0x4000000
 CONFIG_DISTRO_DEFAULTS=y
 CONFIG_BOOTDELAY=3
@@ -27,7 +27,7 @@ CONFIG_DISPLAY_CPUINFO=y
 CONFIG_DISPLAY_BOARDINFO=y
 CONFIG_BOARD_EARLY_INIT_R=y
 CONFIG_MISC_INIT_R=y
-CONFIG_SPL_MAX_SIZE=0x100000
+CONFIG_SPL_MAX_SIZE=0x40000
 CONFIG_SPL_BOARD_INIT=y
 CONFIG_SPL_CACHE=y
 CONFIG_SPL_OPENSBI_SCRATCH_OPTIONS=0x0
@@ -66,4 +66,4 @@ CONFIG_WDT=y
 CONFIG_WDT_ATCWDT200=y
 # CONFIG_BINMAN_FDT is not set
 CONFIG_CONSOLE_RECORD=y
-CONFIG_CONSOLE_RECORD_OUT_SIZE_F=0x2000
\ No newline at end of file
+CONFIG_CONSOLE_RECORD_OUT_SIZE_F=0x2000
-- 
2.34.1

